### PR TITLE
Add component GlobalStylesVariations in @automattic/global-styles

### PR DESCRIPTION
**Pull request by taipeicoder**
## Proposed Changes

This PR refactors the package `@automattic/design-preview` by moving its style variation preview to the package `@automattic/global-styles`. This way, `@automattic/global-styles` becomes the de facto package for rendering style preview for the following pages:

Design preview in the onboarding design picker:
Screenshot 2023-03-27 at 9 54 04 PM

Pattern Assembler.
Screenshot 2023-03-27 at 9 54 36 PM
Screenshot 2023-03-27 at 9 54 44 PM

Theme Detail page.
Screenshot 2023-03-27 at 9 55 30 PM

## Testing Instructions

* Head to the pages mentioned above:
  * Onboarding design picker.
  * Pattern Assembler.
  * Theme Detail page.
* Ensure that the style previews work as before.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/design-preview/src/components/style-variation.tsx
Comment (taipeicoder) [inline packages/design-preview/src/components/style-variation.tsx:1]: This file can also be potentially removed, but decided to keep it for now in order to not make the PR larger than it is.

File: client/my-sites/theme/theme-style-variations/style.scss [hunk 1]
```diff
@@ -14,12 +14,12 @@
 			height: 100%;
 		}
 
-		.theme__sheet-style-variations-previews {
+		.theme__sheet-style-variations-previews .global-styles-variations {
```
Comment (okmttdhr) [inline client/my-sites/theme/theme-style-variations/style.scss]: badge

What is this `.global-styles-variations`?
Do we always need this class to override `@automattic/global-styles`?

**Review by okmttdhr — APPROVED** [review]
✅

File: client/my-sites/theme/theme-style-variations/style.scss (hunk 1 above)
Comment (taipeicoder) [reply client/my-sites/theme/theme-style-variations/style.scss]: `.global-styles-variations` is from `@automattic/global-styles` and it's the wrapper class to all the style previews. In this case, I think  `.theme__sheet-style-variations-previews` is redundant, and we can use `.global-styles-variations` is enough 🙂

Comment (okmttdhr) [reply client/my-sites/theme/theme-style-variations/style.scss]: >.global-styles-variations is from @automattic/global-styles and it's the wrapper class to all the style previews.

Thank you, I understand!

File: packages/global-styles/src/components/global-styles-variations/index.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		[ globalStylesVariations ]
+	);
+
+	return (
+		<GlobalStylesContext.Provider value={ { base: baseGlobalStyles } }>
+			<div className="global-styles-variations">
+				<GlobalStylesVariation
+					key="base"
+					globalStylesVariation={ baseGlobalStyles }
+					isActive={ ! selectedGlobalStylesVariation }
```
Comment (arthur791004) [inline packages/global-styles/src/components/global-styles-variations/index.tsx]: I found the default variation cannot be selected. Maybe we have to use `isDefaultGlobalStyleVariationSlug` here or `onSelect( null )`.

https://user-images.githubusercontent.com/13596067/228115420-56c1b37c-efcd-4da1-8095-9fe4b7ad963f.mov

Comment (taipeicoder) [reply packages/global-styles/src/components/global-styles-variations/index.tsx]: Good catch! For now, I'm supporting both scenarios: (1) when `selectedGlobalStyleVariation` is null or (2) when `selectedGlobalStylesVariation.slug` is default. We can make the behavior more consistent later 🙂

**Review by arthur791004 — APPROVED** [review]
Nice refactor 👍

**Pull request merged**